vendors/tpl-systemes/drivers/birdy-slim-iot/index.js

```diff
@@ -0,0 +1,75 @@
+function asciiToHexa(str){
+    var arr1 = [];
+    for (var n = 0, l = str.length; n < l; n ++) {
+        var hex = Number(str.charCodeAt(n)).toString(16);
+        arr1.push(hex);
+    }
+    return arr1.join('');
+}
+
+function hexaToAscii(hexx) {
+    var hex = hexx.toString();//force conversion
+    var str = '';
+    for (var i = 0; i < hex.length; i += 2)
+        str += String.fromCharCode(parseInt(hex.substr(i, 2), 16));
+    return str;
+}
+
+
+/**
+ * @typedef {Object} DecodedUplink
+ * @property {Object} data - The open JavaScript object representing the decoded uplink payload when no errors occurred
+ * @property {string[]} errors - A list of error messages while decoding the uplink payload
+ * @property {string[]} warnings - A list of warning messages that do not prevent the driver from decoding the uplink payload
+ */
+
+/**
+ * Decode uplink
+ * @param {Object} input - An object provided by the IoT Flow framework
+ * @param {number[]} input.bytes - Array of bytes represented as numbers as it has been sent from the device
+ * @param {number} input.fPort - The Port Field on which the uplink has been sent
+ * @param {Date} input.recvTime - The uplink message time recorded by the LoRaWAN network server
+ * @returns {DecodedUplink} The decoded object
+ */
+function decodeUplink(input) {
+    let result = {
+        message: "",
```

Comment:
result should include "data", "errors", and "warnings". 
fPort and message are not accepted in the lora-alliance signature. 
Take a look at the guide: https://github.com/actility/device-catalog/tree/main/template/sample-vendor/drivers#uplink-decode